torchtune/data/_transforms.py

```diff
@@ -36,18 +38,17 @@ def sharegpt_to_llama2_dialogue(sample: Mapping[str, Any]) -> Dialogue:
     Args:
         sample (Mapping[str, Any]): a single data sample with "conversations" field pointing
             to a list of dict messages.
-
+        train_on_input (bool): whether the prompt should remain unmasked. Default: False
     Returns:
-        Dialogue: a list of messages with "role" and "content" fields. See `torchtune.datasets._types.Message`
+        List[Message]: a list of messages with "role" and "content" fields. See `torchtune.datasets._types.Message`
             and `torchtune.datasets._types.Dialogue` for more details.
     """
     role_map = {"system": "system", "human": "user", "gpt": "assistant"}
     conversations = sample["conversations"]
-
-    dialogue = []
+    messages = []
     for message in conversations:
         role = role_map[message["from"]]
         content = message["value"]
-        dialogue.append(Message(role=role, content=content))
-
-    return dialogue
+        masked = (role != "assistant") and train_on_input
```

Comment:
Hmm, maybe I'm misunderstanding this, but shouldn;t this be

```
masked = (role != "assistant") and not train_on_input
```

i.e. we don't mask if train_on_input is TRUE?

Reply:
Thanks, good catch. I got thrown off by the `test_slimorca_dataset` test, which was always asserting that the final token of labels was equal to eos_id, even when max_seq_len was so short that we were not including the assistant message and train_on_input was False. So I think the test was doing the wrong thing in that case (if train_on_input=False and we only have inputs, then everything should be masked, even the EOS token). Lmk if this makes sense to you